appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockContainer.java

```diff
@@ -256,7 +256,8 @@ protected boolean acceptableSource(DragSource source) {
     if (component instanceof MockVisibleComponent) {
       // Sprites are only allowed on Canvas, not other containers.
       // Map features are only allowed on Map, not other containers.
-      if (!(component instanceof MockSprite) && !(component instanceof MockMapFeature)) {
+      // Chart Data components are only allowed on Charts, not other containers.
+      if (!(component instanceof MockSprite) && !(component instanceof MockMapFeature) && !(component instanceof MockChartData)) {
```

Comment:
Can we refactor this to it's own method, since this is turning into a very complex condition?